SabPicariello/io-app-test
# Flusso cancellazione profilo

# This PR Depends on [#1,#2]

## Short description
Added the functionality to delete the profile through a switch in the new profile page

## List of changes proposed in this pull request
- locales/en/index.yml: added a label for the profile deletion detail.
- locales/it/index.yml:  added a label for the profile deletion detail.
- ts/navigation/ProfileNavigator.tsx: added navigations to the new screens.
- ts/navigation/params/ProfileParamsList.ts: added references to new routes.
- ts/navigation/routes.ts: added references to new routes.
- ts/sagas/testProfile.ts: added function to the testProfile saga to handle response from the delete profile request.
- ts/screens/profile/ProfileTest.tsx: added navigation to the switch showing the deletion status.
- ts/screens/profile/ProfileTestDelete.tsx: first page of the deletion process.
- ts/screens/profile/ProfileTestDeleteDetail.tsx: second page of the deletion process that shows information about data that will be deleted.

## How to test

1) Set the environment to dev, run application, open the new profile page and make sure you have not requested the profile deletion (the dedicated switch is 'OFF').
2) Turn ON the switch that shows the profile deletion status to start the deletion process.
3) The first Screen (`ProfileTestDelete`) will be rendered with an InfoBox. Verify that Cancel button leads to the previous page and Continue button leads to the next page (`ProfileTestDeleteDetail`).
4) When the screen `ProfileTestDeleteDetail` is rendered, verify that the informations are correctly displayed, Cancel button leads to the previous page and Continue button allows you to submit the cancellation request.
5) Delete the request and verify that, by making the server generate an error, an `Alert` shows up with a retry button.


------

To cancel a deletion request you can send a `DELETE` reqeust to dev-server:
url : `/user-data-processing/DELETE`
